packages/epo-widget-lib/src/widgets/CameraFilter/SpectrumDisplay/ColorLabels/ColorLabels.tsx

```diff
@@ -0,0 +1,96 @@
+import { intersection } from "@/lib/utils";
+import { FunctionComponent } from "react";
+import { VisibleColor } from "../../data";
+import * as Styled from "../../styles";
+
+interface ColorLabelsProps {
+  colors: VisibleColor[];
+  activeRange: number[];
+  isBandSelected: boolean;
+  isCondensed: boolean;
+}
+
+const ColorLabels: FunctionComponent<ColorLabelsProps> = ({
+  colors,
+  activeRange,
+  isBandSelected,
+  isCondensed,
+}) => {
+  const colorsMin = isCondensed ? 350 : colors[0].range[0];
+  const colorsMax = isCondensed ? 1000 : 820;
+  const totalRange = colorsMax - colorsMin;
+  const labelDistance = Math.floor(totalRange / (colors.length - 1));
+  return (
+    <g role="list" data-testid="color-labels">
+      {colors.map(({ name, range, color }, i) => {
+        const textPosition = colorsMin + labelDistance * i;
+        const isHidden =
+          isBandSelected && intersection(range, activeRange) === null;
```

Comment:
hidden from the accessibility tree if there is a band selected and no intersection between the current color and the selected spectrum range